plugins/jenkins-backend/src/service/router.ts

```diff
@@ -34,13 +37,26 @@ export interface RouterOptions {
   logger: Logger;
   jenkinsInfoProvider: JenkinsInfoProvider;
   permissions?: PermissionEvaluator | PermissionAuthorizer;
+  identity?: IdentityApi;
 }
 
 /** @public */
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { jenkinsInfoProvider, permissions, logger } = options;
+  const { jenkinsInfoProvider, permissions, logger, identity } = options;
+
+  async function getRequestToken(
+    request: express.Request,
+  ): Promise<string | undefined> {
+    if (identity) {
+      const user = await identity.getIdentity({ request });
```

Comment:
Noting that this general pattern is somewhat tricky to roll out right now due to the token now being required. We likely need some form of control for whether identities are required or not, and I'm not quite sure who should take that decision. In the end I feel we need to figure out all of service-to-service auth, including stuff like CI tokens, as well as how that mixes with user auth, and also introducing authentication across the entire backend, except the login page. Not something we can put prio on right now.

The reason this is gonna end up being a bit tricky is that right now you've got this manual wiring where you can decide whether you want to pass in the identity API or not, but in the new backend system it'll be provided automatically. So at that point we'd need to figure out how to handle optional auth.

What do you think? Could this be something like `identity.getIdentity({ request, optional: true })`?

Reply:
Yeah that sounds like a good approach. I like it. I am on a week break this week. But maybe one of my team can take it or I'll take care of it next week.

Reply:
I had a little think about this. I am not finding an obvious way to add that optional parameter in a way that is backwards compatible. Maybe there is some trick I am missing.

Reply:
If it's optional then you should be able to simply add it. It's a bit of a breaking change for integrators, but our versioning policy states that the app/backend packages should always be at or ahead of the plugins, so anyone using the default implementation is alright. If there are custom implementations of the API then they'll miss out on the optionality support, but it's in an area were we're alright with not calling it a breaking change.

Reply:
https://github.com/backstage/backstage/pull/14533 - here is a first pass patchset.